batch2/batch/driver/scheduler.py

```diff
@@ -19,6 +20,59 @@ def __init__(self, app):
         asyncio.ensure_future(self.loop('cancel_loop', self.cancel_state_changed, self.cancel_1))
         asyncio.ensure_future(self.bump_loop())
 
+    async def compute_fair_share(self):
+        free_cores_mcpu = sum([worker.free_cores_mcpu for worker in self.inst_pool.healthy_instances_by_free_cores])
+
+        sorted_running_cores_mcpu = sortedcontainers.SortedSet(key=lambda x: x[1])
+        sorted_total_cores_mcpu = sortedcontainers.SortedSet(key=lambda x: x[1])
+
+        running_cores_mcpu = {}
+        allocated_cores = {}
```

Comment:
Few readability comments:
 - I like key_value for dicts, so: user_running_cores_mcpu and users_allocated_cores, say
 - If you declare the dictionaries first, you can use them in the sorted sets, like:

 > pending_users_by_running_cores_mcpu = SortedSet(key=lambda user: user_running_cores_mcpu[user])